internal/api/v1/configurationbinding/deletebinding.go

```diff
@@ -10,7 +10,7 @@ import (
 	apierror "github.com/epinio/epinio/pkg/api/core/v1/errors"
 )
 
-func DeleteBinding(ctx context.Context, cluster *kubernetes.Cluster, namespace, appName, configurationName, username string) apierror.APIErrors {
+func DeleteBinding(ctx context.Context, cluster *kubernetes.Cluster, namespace, appName, configurationName, username string, fromService bool) apierror.APIErrors {
```

Comment:
In my opinion, this function has too many responsibilities. I think it shouldn't be responsible to check if the binding should be deleted or not (checking the origin and this additional boolean flag here). It should simply delete the binding. The caller should be responsible to decide whether such a deletion should happen or not.

The caller will need to load the configuration object to be able to check the `config.Origin` value of course but that loading won't need to happen again here, since this is only used for that check (which will now live elsewhere). Even if the configuration object was needed here, we could as well pass it instead of passing just the configuration name.

The reason for my suggestion is:

- keep this function to a single responsibility
- avoid the boolean parameter to keep the code more readable

(relevant random google result: https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/)

Reply:
Will look into refactoring the code.

Reply:
Commit 4032525db1f69d2e54eaca68e50a578efe37adf9